spec/helpers/mailing_list/unsubscribe_url_spec.rb

```diff
@@ -0,0 +1,49 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+#  Copyright (c) 2021, Hitobito AG. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+require 'spec_helper'
+
+describe MailingList::UnsubscribeUrl do
+  let(:mailing_list) { mailing_lists(:leaders) }
+  let(:link) { MailingList::UnsubscribeUrl.unsubscribe_link(mailing_list) }
+  let(:link_with_html) { MailingList::UnsubscribeUrl.unsubscribe_link(mailing_list, html: true) }
+
+  context 'protocol' do
+    protocol_with_ssl = 'https://'
+    protocol_without_ssl = 'http://'
+
+    it 'should include https' do
+      expect(Rails.application.config).to receive(:force_ssl).and_return(true)
+      expect(link).to include(protocol_with_ssl)
+    end
+
+    it 'should not include https' do
+      expect(link).to include(protocol_without_ssl)
+    end
+  end
+
+  context 'clickable link' do
+    anchor_tag = '<a href='
+
+    it 'should be clickable' do
+      expect(link_with_html).to include(anchor_tag)
+    end
+
+    it 'should be text' do
+      expect(link).not_to include(anchor_tag)
+    end
+  end
+
+  context 'host' do
+    it 'should validate to be env host' do
+      host = ENV.fetch('RAILS_HOST_NAME', 'localhost:3000')
+      expect(link).to include(host)
```

Comment:
```suggestion
      expect(link).to include('localhost:3000')
```

another test to ensure RAILS_HOST_NAME is concerned